==> diffs/here_doc.diff <==
diff of execution of:
echo 'cat << EOF;hi;ha;ho;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:54.391632902 +0200
+++ found	2024-09-05 15:28:54.410633365 +0200
@@ -1,8 +1,11 @@
 hi
 ha
 ho
 
 ======== stderr =======
-
+[progname]: hi: command not found
+[progname]: ha: command not found
+[progname]: ho: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/here_doc_multivar.diff <==
diff of execution of:
echo 'cat <<EOF; $PWD and $OLDPWD;ha;$PATH ;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:54.520636047 +0200
+++ found	2024-09-05 15:28:54.546636681 +0200
@@ -1,8 +1,11 @@
  /home/bruno/Github42/Minishell/test/temp and /home/bruno/Github42/Minishell/test
 ha
 /home/bruno/.local/bin:/home/bruno/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/bruno/.local/bin:/home/bruno/.local/bin:/home/bruno/.local/bin:/home/bruno/.local/bin 
 
 ======== stderr =======
-
+[progname]: /home/bruno/Github42/Minishell/test/temp: Is a directory
+[progname]: ha: command not found
+[progname]: /home/bruno/.local/bin:/home/bruno/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/bruno/.local/bin:/home/bruno/.local/bin:/home/bruno/.local/bin:/home/bruno/.local/bin: No such file or directory
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/here_doc_quotes.diff <==
diff of execution of:
echo 'cat << EOF;\'$PWD\';EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:54.476634975 +0200
+++ found	2024-09-05 15:28:54.495635437 +0200
@@ -1,6 +1,7 @@
 '/home/bruno/Github42/Minishell/test/temp'
 
 ======== stderr =======
-
+[progname]: $PWD: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/here_doc_status1.diff <==
diff of execution of:
echo 'cat << EOF;$?;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:54.637638899 +0200
+++ found	2024-09-05 15:28:54.656639363 +0200
@@ -1,6 +1,7 @@
 0
 
 ======== stderr =======
-
+[progname]: 0: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/here_doc_status2.diff <==
diff of execution of:
echo 'cat fail; cat << EOF;$?;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:54.684640045 +0200
+++ found	2024-09-05 15:28:54.706640582 +0200
@@ -1,6 +1,8 @@
 1
 
 ======== stderr =======
 [progname]: fail: No such file or directory
+[progname]: 0: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/here_doc_var.diff <==
diff of execution of:
echo 'cat << EOF;$PWD;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:54.437634023 +0200
+++ found	2024-09-05 15:28:54.455634462 +0200
@@ -1,6 +1,7 @@
 /home/bruno/Github42/Minishell/test/temp
 
 ======== stderr =======
-
+[progname]: /home/bruno/Github42/Minishell/test/temp: Is a directory
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/path_add_local.diff <==
diff of execution of:
echo 'export PATH=$PATH\':./\'; echo $PATH; infile' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:56.626687304 +0200
+++ found	2024-09-05 15:28:56.642687693 +0200
@@ -1,6 +1,6 @@
 /home/bruno/.local/bin:/home/bruno/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/bruno/.local/bin:/home/bruno/.local/bin:/home/bruno/.local/bin:/home/bruno/.local/bin:./
 
 ======== stderr =======
-[progname]: infile: Permission denied
+[progname]: infile: command not found
 ======== return =======
 return value: 0

==> diffs/path_empty_part_local.diff <==
diff of execution of:
echo 'export PATH=$PATH\':::::/hehe\'; echo $PATH; infile' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:56.898693914 +0200
+++ found	2024-09-05 15:28:56.914694303 +0200
@@ -1,6 +1,6 @@
 /home/bruno/.local/bin:/home/bruno/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/bruno/.local/bin:/home/bruno/.local/bin:/home/bruno/.local/bin:/home/bruno/.local/bin:::::/hehe
 
 ======== stderr =======
-[progname]: infile: Permission denied
+[progname]: infile: command not found
 ======== return =======
 return value: 0

==> diffs/questionable_exec_dir.diff <==
diff of execution of:
echo 'export PATH="$PATH:$PWD"; mkdir asdfg; asdfg' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:58.570734464 +0200
+++ found	2024-09-05 15:28:58.589734924 +0200
@@ -1,5 +1,6 @@
 
 ======== stderr =======
+[progname]: mkdir: command not found
 [progname]: asdfg: command not found
 ======== return =======
 return value: 1

==> diffs/questionable_exec_dot.diff <==
diff of execution of:
echo '.' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:58.457731727 +0200
+++ found	2024-09-05 15:28:58.471732066 +0200
@@ -1,6 +1,5 @@
 
 ======== stderr =======
-[progname]: .: filename argument required
-[progname]: usage: . filename [arguments]
+[progname]: .: Is a directory
 ======== return =======
 return value: 1

==> diffs/questionable_exec_dotdot.diff <==
diff of execution of:
echo '..' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:58.494732623 +0200
+++ found	2024-09-05 15:28:58.508732962 +0200
@@ -1,5 +1,5 @@
 
 ======== stderr =======
-[progname]: ..: command not found
+[progname]: ..: Is a directory
 ======== return =======
 return value: 1

==> diffs/questionable_export_spaces1.diff <==
diff of execution of:
echo 'export a=\'    \'; echo $a' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:58.272727245 +0200
+++ found	2024-09-05 15:28:58.287727608 +0200
@@ -1,6 +1,6 @@
-
+    
 
 ======== stderr =======
 
 ======== return =======
 return value: 0

==> diffs/questionable_export_spaces2.diff <==
diff of execution of:
echo 'export a="a    b"; echo $a' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:58.307728093 +0200
+++ found	2024-09-05 15:28:58.322728456 +0200
@@ -1,6 +1,6 @@
-a b
+a    b
 
 ======== stderr =======
 
 ======== return =======
 return value: 0

==> diffs/questionable_glob_order.diff <==
diff of execution of:
echo 'touch a b c A B C aa ab ac Aa Ab Ac aA aB aC; echo *' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:58.378729813 +0200
+++ found	2024-09-05 15:28:58.396730249 +0200
@@ -1,6 +1,6 @@
-a A aa aA Aa ab aB Ab ac aC Ac b B c C infile readonly
+aC infile .. Aa B ac C A readonly Ab ab aB aa aA b c Ac a .
 
 ======== stderr =======
 
 ======== return =======
 return value: 0

==> diffs/questionable_pwd_rmdir.diff <==
diff of execution of:
echo 'mkdir tmp; cd tmp; rmdir $PWD; pwd' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:58.667736814 +0200
+++ found	2024-09-05 15:28:58.686737274 +0200
@@ -1,6 +1,5 @@
-/home/bruno/Github42/Minishell/test/temp/tmp
 
 ======== stderr =======
-
+getcwd() error
 ======== return =======
-return value: 0
+return value: 1

==> diffs/questionable_redir_ambiguous.diff <==
diff of execution of:
echo 'cat < *;' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:58.418730782 +0200
+++ found	2024-09-05 15:28:58.435731194 +0200
@@ -1,5 +1,6 @@
 
 ======== stderr =======
-[progname]: *: ambiguous redirect
+[progname]: ..: Is a directory
+[progname]: .: Is a directory
 ======== return =======
 return value: 1

==> diffs/redir_heredoc2.diff <==
diff of execution of:
echo 'grep o << EOF > out;hello;there;EOF; ls -l --time-style=+' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:54.348631854 +0200
+++ found	2024-09-05 15:28:54.370632390 +0200
@@ -1,9 +1,11 @@
 total 8
 -rw-rw-r-- 1 bruno bruno 34  infile
 -rw-rw-r-- 1 bruno bruno  6  out
 -r--r--r-- 1 bruno bruno  0  readonly
 
 ======== stderr =======
-
+[progname]: hello: command not found
+[progname]: there: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/redir_heredoc.diff <==
diff of execution of:
echo 'cat << EOF;hello;there;EOF' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:54.303630757 +0200
+++ found	2024-09-05 15:28:54.322631220 +0200
@@ -1,7 +1,9 @@
 hello
 there
 
 ======== stderr =======
-
+[progname]: hello: command not found
+[progname]: there: command not found
+[progname]: EOF: command not found
 ======== return =======
 return value: 0

==> diffs/redir_multi_here.diff <==
diff of execution of:
echo 'cat <<EOF <<OH;i;am;here;EOF;oh;OH' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:55.539660870 +0200
+++ found	2024-09-05 15:28:55.561661405 +0200
@@ -1,6 +1,11 @@
 oh
 
 ======== stderr =======
-
+[progname]: i: command not found
+[progname]: am: command not found
+[progname]: here: command not found
+[progname]: EOF: command not found
+[progname]: oh: command not found
+[progname]: OH: command not found
 ======== return =======
 return value: 0

==> diffs/return_ko2.diff <==
diff of execution of:
echo 'echo a > readonly; echo $?' | tr ';' '\n' | [shell]
--- expected	2024-09-05 15:28:57.861717284 +0200
+++ found	2024-09-05 15:28:57.878717696 +0200
@@ -1,6 +1,7 @@
-1
+a
+0
 
 ======== stderr =======
 [progname]: readonly: Permission denied
 ======== return =======
 return value: 0
